### PR TITLE
ENH: override sf for Pearson3 distribution

### PR DESCRIPTION
**Pull request by OmarManzoor**
#### Reference issue

Towards: gh-17832

#### What does this implement/fix?

- Adds and overrides the survival function for the Pearson3 distribution
- Adds a reference distribution for Pearson3
- Adds sets based on the reference distribution

#### Additional information

cc: @mdhaber

### CODE REVIEW

**Comment by OmarManzoor** [comment]
image

Code
<Details>

```python
from scipy import stats
import numpy as np
import mpmath
from mpmath import mp
import matplotlib.pyplot as plt
rng = np.random.default_rng()
mp.dps = 500

def _preprocess(x, skew):
    loc = 0.0
    scale = 1.0
    norm2pearson_transition = 0.000016

    ans, x, skew = np.broadcast_arrays(1.0, x, skew)
    ans = ans.copy()

    mask = np.absolute(skew) < norm2pearson_transition
    invmask = ~mask

    beta = 2.0 / (skew[invmask] * scale)
    alpha = (scale * beta)**2
    zeta = loc - alpha / beta

    transx = beta * (x[invmask] - zeta)
    return ans, x, transx, mask, invmask, beta, alpha, zeta

def pearson3_sf(x, skew):
    ans, x, transx, mask, invmask, _, alpha, _ = _preprocess(x, skew)
    
    ans[mask] = stats.norm._sf(x[mask])

    skew = np.broadcast_to(skew, invmask.shape)
    invmask1a = np.logical_and(invmask, skew > 0)
    invmask1b = skew[invmask] > 0
    ans[invmask1a] = stats.gamma.sf(transx[invmask1b], alpha[invmask1b])

    invmask2a = np.logical_and(invmask, skew < 0)
    invmask2b = skew[invmask] < 0
    ans[invmask2a] = stats.gamma.cdf(transx[invmask2b], alpha[invmask2b])

    return ans

skew = 0.1
pearson3_mp = Pearson3(skew=skew)
x = np.logspace(-2, 3)
mpmath_values = np.array([pearson3_mp.sf(_x) for _x in x], np.float64)
plt.loglog(x, stats.pearson3.sf(x, skew), label="pearson3 main", ls="dashed")
plt.loglog(x, pearson3_sf(x, skew), label="pearson3 pr", ls="dashdot")
plt.loglog(x, mpmath_values, label="pearson3 mpmath", ls="dotted")
plt.legend()
plt.show()
```

</Details>

File: scipy/stats/_continuous_distns.py [hunk 1]
```diff
@@ -7713,6 +7713,23 @@ def _cdf(self, x, skew):
 
         return ans
 
+    def _sf(self, x, skew):
```
Comment (mdhaber) [inline scipy/stats/_continuous_distns.py:7716]: Copying the structure of the CDF here is fine with me.

File: scipy/stats/tests/test_distributions.py [hunk 2]
```diff
@@ -2329,6 +2329,17 @@ def test_ppf_bug_17050(self):
         assert_allclose(stats.pearson3.ppf(x, skew),
                         -stats.pearson3.isf(x, -skew))
 
+    def test_sf(self):
+        # reference values were computed via the reference distribution, e.g.
+        # mp.dps = 50; Pearson3(skew=skew).sf(x). Check positive, negative,
+        # and zero skew due to branching.
```
Comment (mdhaber) [inline scipy/stats/tests/test_distributions.py:2335]: I added tests for the other branches of the `_sf` implementation.

File: scipy/stats/tests/test_generation/reference_distributions.py [hunk 3]
```diff
@@ -333,6 +333,18 @@ def _pdf(self, x, alpha, beta):
         return a * q**-1 * mp.besselk(1, alpha*q) * mp.exp(beta*x)
 
 
+class Pearson3(ReferenceDistribution):
```
Comment (mdhaber) [inline scipy/stats/tests/test_generation/reference_distributions.py:336]: The intent of the reference distribution is to be simple and easy to verify. I went ahead and pushed a simplification.

**Review by mdhaber — APPROVED** [review]

File: scipy/stats/tests/test_generation/reference_distributions.py (hunk 3 above)
Comment (OmarManzoor) [reply scipy/stats/tests/test_generation/reference_distributions.py:336]: Is this a variation of the formula for the pdf?

Comment (mdhaber) [reply scipy/stats/tests/test_generation/reference_distributions.py:336]: It is the PDF as written in the docstring.

Comment (OmarManzoor) [reply scipy/stats/tests/test_generation/reference_distributions.py:336]: Thank you.

**Pull request merged**